wasm/jit/engine.go

```diff
@@ -240,9 +170,8 @@ const initialStackSize = 1024
 
 func newEngine() *engine {
 	e := &engine{
-		stack:                     make([]uint64, initialStackSize),
-		compiledWasmFunctionIndex: make(map[*wasm.FunctionInstance]int64),
-		compiledHostFunctionIndex: make(map[*wasm.FunctionInstance]int64),
+		stack:             make([]uint64, initialStackSize),
+		compiledFunctions: make(map[wasm.FunctionAddress]*compiledFunction),
```

Comment:
food for thought. later we may end up needing another package for things that aren't necessarily wasm related, such as a function address.

Reply:
`FunctionAddress` is a Wasm concept, and this is different from the actual memory address in a physical machine! I added a  doc on wasm.FunctionAddress in store.go